### PR TITLE
homework_1 first attempt

### PR DESCRIPTION
**Pull request by IliaRusecki**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,141 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>com.epam.tat.module4</groupId>
+            <artifactId>calculator</artifactId>
+            <version>1.0</version>
+        </dependency>
+    </dependencies>
+
+        <repositories>
```
Comment (khda91) [inline pom.xml]: What is the purpose of using `repositories` here?

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,141 @@
...
+                    <forkCount>0</forkCount>
+                    <suiteXmlFiles>
+                        <suiteXmlFile>${suite.file}</suiteXmlFile>
+                    </suiteXmlFiles>
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-install-plugin</artifactId>
```
Comment (khda91) [inline pom.xml]: Why do you need this plugin?

File: src/test/java/hw1/dataproviders/SubtractDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.testng.annotations.DataProvider;
+
+public class SubtractDataProvider {
+
+
+    @DataProvider(name = "SubtractLong Data Provider")
+    public Object[][] subtractLongTestDataProvider() {
+        return new Object[][]{
+                {1, 2, -1},
+                {-9223372036854775808L, 1, 9223372036854775807L},
```
Comment (khda91) [inline src/test/java/hw1/dataproviders/SubtractDataProvider.java]: What are these values means?

File: src/test/java/hw1/testclasses/AbstractBaseTest.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package hw1.testclasses;
+
+import com.epam.tat.module4.Calculator;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+public class AbstractBaseTest {
+    Calculator calculator = null;
```
Comment (khda91) [inline src/test/java/hw1/testclasses/AbstractBaseTest.java]: Should be empty line

File: src/test/java/hw1/testclasses/DivideTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+public class DivideTest extends AbstractBaseTest {
+
+    @Test(dataProviderClass = DivideDataProvider.class,
+            dataProvider = "DivideLong Data Provider",
+            groups = "Multiply and Divide")
+    public void divideLongTest(long a, long b, long expected) {
+        long actual = calculator.div(a, b);
+        System.out.println("DivideLongTest");
+        Assert.assertEquals(actual, expected);
```
Comment (khda91) [inline src/test/java/hw1/testclasses/DivideTest.java]: It is better to use `static import` for the `Assert` methods.

File: pom.xml (hunk 1 above)
Comment (IliaRusecki) [reply pom.xml]: Yeah, actually there's no purpose... I didn't now how to attach a jar file to the project and used some script from a StackOverflow... but now it seem's that  I found out how to do it, I fixed it now.

File: pom.xml (hunk 2 above)
Comment (IliaRusecki) [reply pom.xml]: The same thing as in privious quastion

File: src/test/java/hw1/dataproviders/SubtractDataProvider.java (hunk 3 above)
Comment (IliaRusecki) [reply src/test/java/hw1/dataproviders/SubtractDataProvider.java]: Here  I just wanted to look what's gonna be if I add or subtract a number to or from the max or min of the lang veraible and it worked as it should.. when subtract 1 from the min of lang it's turning to the max value.... but, yeah it looks strange, I've changed it to normal data!

File: src/test/java/hw1/testclasses/AbstractBaseTest.java (hunk 4 above)
Comment (IliaRusecki) [reply src/test/java/hw1/testclasses/AbstractBaseTest.java]: So, i guess in this small project there's no need to use Abstract class, so I deleted it

File: src/test/java/hw1/testclasses/DivideTest.java (hunk 5 above)
Comment (IliaRusecki) [reply src/test/java/hw1/testclasses/DivideTest.java]: Ok, thenks, I'll change it

**Pull request merged**